crates/router/src/core/payment_link.rs

```diff
@@ -0,0 +1,834 @@
+};
+
+pub async fn retrieve_payment_link(
+    state: AppState,
+    merchant_account: domain::MerchantAccount,
+    payment_link_id: String,
+) -> RouterResponse<api_models::payments::RetrievePaymentLinkResponse> {
+    let db = &*state.store;
+    let payment_link_object = db
+        .find_payment_link_by_payment_link_id(&payment_link_id, &merchant_account.merchant_id)
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::PaymentLinkNotFound)?;
+
+    let response = api_models::payments::RetrievePaymentLinkResponse {
+        payment_link_id: payment_link_object.payment_link_id,
+        payment_id: payment_link_object.payment_id,
+        merchant_id: payment_link_object.merchant_id,
+        link_to_pay: payment_link_object.link_to_pay,
+        amount: payment_link_object.amount,
+        currency: payment_link_object.currency,
+        created_at: payment_link_object.created_at,
+        last_modified_at: payment_link_object.last_modified_at,
+        link_expiry: payment_link_object.fullfilment_time,
+    };
+
+    Ok(services::ApplicationResponse::Json(response))
+}
+
+pub async fn intiate_payment_link_flow(
+    state: AppState,
+    merchant_account: domain::MerchantAccount,
+    merchant_id: String,
+    payment_id: String,
+) -> RouterResponse<services::PaymentLinkFormData> {
+    let db = &*state.store;
+    let payment_intent = db
+        .find_payment_intent_by_payment_id_merchant_id(
+            &payment_id,
+            &merchant_id,
+            merchant_account.storage_scheme,
+        )
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::PaymentNotFound)?;
+
+    helpers::validate_payment_status_against_not_allowed_statuses(
+        &payment_intent.status,
+        &[
+            storage_enums::IntentStatus::Cancelled,
+            storage_enums::IntentStatus::Succeeded,
+            storage_enums::IntentStatus::Processing,
+            storage_enums::IntentStatus::RequiresCapture,
+            storage_enums::IntentStatus::RequiresMerchantAction,
+        ],
+        "create",
+    )?;
+
+    let body = get_html_body();
+    let css_script = get_css();
+    let js_script = get_js_script(
+        payment_intent.amount.to_string(),
+        payment_intent.currency.unwrap_or_default().to_string(),
+        merchant_account.publishable_key.unwrap_or_default(),
+        payment_intent.client_secret.unwrap_or_default(),
+        payment_intent.payment_id,
+    );
+
+    let payment_link_data = services::PaymentLinkFormData {
+        js_script,
+        css_script,
+        body,
+        base_url: state.conf.server.base_url.clone(),
+    };
+    Ok(services::ApplicationResponse::PaymenkLinkForm(Box::new(
+        payment_link_data,
+    )))
+}
+
+fn get_js_script(
+    amount: String,
+    currency: String,
+    pub_key: String,
+    secret: String,
+    payment_id: String,
+) -> String {
+    format!(
+    "
+    <script>
+        window.__PAYMENT_DETAILS_STR = JSON.stringify({{
```

Comment:
Move this outside to a file which you can read on startup? Otherwise minor css changes too will require a full re compile.